sopel/cli/run.py

```diff
@@ -0,0 +1,658 @@
+            if hasattr(signal, 'SIGUSR2'):
+                signal.signal(signal.SIGUSR2, signal_handler)
+            if hasattr(signal, 'SIGILL'):
+                signal.signal(signal.SIGILL, signal_handler)
+            logger.setup_logging(p)
+            p.run(config.core.host, int(config.core.port))
+        except KeyboardInterrupt:
+            break
+        except Exception:  # TODO: Be specific
+            trace = traceback.format_exc()
+            try:
+                tools.stderr(trace)
+            except Exception:  # TODO: Be specific
+                pass
+            logfile = open(os.path.join(config.core.logdir, 'exceptions.log'), 'a')
+            logfile.write('Critical exception in core')
+            logfile.write(trace)
+            logfile.write('----------------------------------------\n\n')
+            logfile.close()
+            # TODO: This should be handled by command_start
+            # All we should need here is a return value, but replacing the
+            # os._exit() call below (at the end) broke ^C.
+            # This one is much harder to test, so until that one's sorted it
+            # isn't worth the risk of trying to remove this one.
+            os.unlink(pid_file)
+            os._exit(1)
+
+        if not isinstance(delay, int):
+            break
+        if p.wantsrestart:
+            return -1
+        if p.hasquit:
+            break
+        tools.stderr(
+            'Warning: Disconnected. Reconnecting in %s seconds...' % delay)
+        time.sleep(delay)
+    # TODO: This should be handled by command_start
+    # All we should need here is a return value, but making this
+    # a return makes Sopel hang on ^C after it says "Closed!"
+    os.unlink(pid_file)
+    os._exit(0)
+
+
+def add_legacy_options(parser):
+    parser.add_argument("-d", '--fork', action="store_true",
+                        dest="daemonize", help="Daemonize Sopel")
+    parser.add_argument("-q", '--quit', action="store_true", dest="quit",
+                        help=(
+                            "Gracefully quit Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop` instead)"))
+    parser.add_argument("-k", '--kill', action="store_true", dest="kill",
+                        help=(
+                            "Kill Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop --kill` instead)"))
+    parser.add_argument("-r", '--restart', action="store_true", dest="restart",
+                        help=(
+                            "Restart Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel restart` instead)"))
+    parser.add_argument("-l", '--list', action="store_true",
+                        dest="list_configs",
+                        help="List all config files found")
```

Comment:
Ditto here as for `--fork`/`-d`, but I can't find any reference to this arg outside `command_legacy`.

Will this become part of `start` for Sopel 8, when `legacy` goes away? (`# TODO`?)

Reply:
Look at `get_configuration`, it's in here:

    bot_config._is_daemonized = options.daemonize
    return bot_config

Reply:
(nevermind the previous comment, wrong one)

For now,  -l/--list exists from `sopel` only. However, I would like to move this option to a separated command, ie. `sopel-config`, with this interface:

    $ sopel-config init
    $ sopel-config list
    $ sopel-config rename

And other features.

Cherry on top: `sopel` is like `git`, where `git commit` is, in fact, `git-commit`. It means that `sopel config` and `sopel-config` will be the same thing, code-wise.

I'm confident enough that this will be taken care of soon enough for us not to add extra TODO.

Reply:
Fair enough.

And even if it doesn't get done (long shot, because you're on top of your shit) it will be quite obvious that this needs replacing when it's time to remove the legacy parser, because this will be one of very few options without a deprecation notice.